### PR TITLE
Images 2 (I hate blobstore!!) 

### PR DESCRIPTION
**Pull request by hedayaw**

### CODE REVIEW

**Comment by RickyMarin** [comment]
I can't open the imageURL per an incompatibility with Java11 so I am assigning this to Jeremy to see if he can find anything

**Comment by hedayaw** [comment]
Maybe we have different versions of Java? I tested it on my laptop and everything works and I'm pretty sure I have Java 11 but I can double check.

**Comment by RickyMarin** [comment]
Jeremy and I were going through it yesterday and apparently, there is an issue that even Kevin doesn't even know how to solve.

**Comment by fatlotus** [comment]
Looks like there might be merge conflicts. GitHub has a new edit panel that seems like it might help.

I tried it locally and ran into a bug (I'm going to try again after lunch):

```
org.eclipse.jetty.servlet.ServletHolder$1: java.lang.IllegalStateException: Must call one of set*BlobStorage() first.
	at org.eclipse.jetty.servlet.ServletHolder.makeUnavailable(ServletHolder.java:597)
	at org.eclipse.jetty.servlet.ServletHolder.initServlet(ServletHolder.java:661)
	at org.eclipse.jetty.servlet.ServletHolder.getServlet(ServletHolder.java:499)
	at org.eclipse.jetty.servlet.ServletHolder.ensureInstance(ServletHolder.java:791)
	at org.eclipse.jetty.servlet.ServletHolder.prepare(ServletHolder.java:776)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:579)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:524)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:226)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1180)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:512)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1112)
	at com.google.appengine.tools.development.jetty9.DevAppEngineWebAppContext.doScope(DevAppEngineWebAppContext.java:94)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134)
	at com.google.appengine.tools.development.jetty9.JettyContainerService$ApiProxyHandler.handle(JettyContainerService.java:597)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134)
	at org.eclipse.jetty.server.Server.handle(Server.java:534)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:320)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:251)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:283)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:108)
	at org.eclipse.jetty.io.SelectChannelEndPoint$2.run(SelectChannelEndPoint.java:93)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:303)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:136)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalStateException: Must call one of set*BlobStorage() first.
	at com.google.appengine.api.blobstore.dev.BlobStorageFactory.getBlobStorage(BlobStorageFactory.java:24)
	at com.google.appengine.api.blobstore.dev.UploadBlobServlet.init(UploadBlobServlet.java:102)
	at javax.servlet.GenericServlet.init(GenericServlet.java:244)
	at org.eclipse.jetty.servlet.ServletHolder.initServlet(ServletHolder.java:643)
	... 28 more
```

**Comment by fatlotus** [comment]
Hmm, I'm still not sure what this means, but as best I can tell, it's just an issue with testing locally?

I'll look over the code and see if it seems well formed.

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 1]
```diff
@@ -39,6 +39,13 @@ public void storeMessage(Message message) {
     messageEntity.setProperty("timestamp", message.getTimestamp());
     messageEntity.setProperty("sentimentScore", message.getSentimentScore());
     messageEntity.setProperty("location", message.getLocation());
+
+    //added by Hedaya
```
Comment (fatlotus) [inline src/main/java/com/google/codeu/data/Datastore.java:43]: nitpick: Don't need these comments

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 2]
```diff
@@ -75,7 +82,9 @@ public int getTotalMessageCount(){
         * Old messages get score of 0
         */
         float sentimentScore = entity.getProperty("sentimentScore") == null? (float) 0.0 : ((Double) entity.getProperty("sentimentScore")).floatValue();
+        String imageUrl = (String) entity.getProperty("imageUrl");
         Message message = new Message(id, user, text, timestamp, sentimentScore, location);
+        message.setImageUrl(imageUrl);
```
Comment (fatlotus) [inline src/main/java/com/google/codeu/data/Datastore.java:87]: Might want to verify that `imageUrl` is not null first.

File: src/main/java/com/google/codeu/data/Message.java [hunk 3]
```diff
@@ -45,6 +46,16 @@ public Message(UUID id, String user, String text, long timestamp, float sentimen
     this.location = location;
   }
 
+  //added by Hedaya
```
Comment (fatlotus) [inline src/main/java/com/google/codeu/data/Message.java:48]: nitpick: probably don't need these comments either (the version history will have this information).

**Review by fatlotus — APPROVED** [review]

**Pull request merged**